python/dgl/transform.py

```diff
@@ -1991,13 +2000,32 @@ def to_simple(g,
         node_frames = utils.extract_node_subframes(g, None)
         utils.set_new_frames(simple_graph, node_frames=node_frames)
     if copy_edata:
-        eids = []
-        for i in range(len(g.canonical_etypes)):
-            feat_idx = F.asnumpy(edge_maps[i])
-            _, indices = np.unique(feat_idx, return_index=True)
-            eids.append(F.zerocopy_from_numpy(indices))
-
-        edge_frames = utils.extract_edge_subframes(g, eids)
+        if aggregator == 'random':
+            eids = []
+            for i in range(len(g.canonical_etypes)):
+                feat_idx = F.asnumpy(edge_maps[i])
+                _, indices = np.unique(feat_idx, return_index=True)
+                eids.append(F.zerocopy_from_numpy(indices))
+
+            edge_frames = utils.extract_edge_subframes(g, eids)
+        else:  # 'sum'/'mean'
+            edge_frames = []
+            for i in range(len(g.canonical_etypes)):
+                feat_idx = edge_maps[i]
+                _, indices = np.unique(F.asnumpy(feat_idx), return_index=True)
+                _num_rows = len(indices)
+                _data = {}
+                for key, col in g._edge_frames[i]._columns.items():
+                    data = col.data
+                    new_data = F.scatter_add(data, feat_idx, _num_rows)
+                    if aggregator == 'mean':
+                        norm = F.astype(counts[i], F.dtype(data))
+                        norm = F.reshape(norm, (F.shape(norm)[0],) + (1,) * (F.ndim(data) - 1))
+                        new_data /= norm
+                    _data[key] = new_data
+
+                newf = Frame(data=_data, num_rows=_num_rows)
+                edge_frames.append(newf)
```

Comment:
Can we functionize each of the branches?

Reply:
yep